### PR TITLE
[cli] [grpc{tabletconn,tmclient}] Migrate flags to `pflag`

### LINKED ISSUE #10974: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/grpctabletconn`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/grpctabletconn/*.go
go/vt/vttablet/grpctabletconn/conn.go:  cert = flag.String("tablet_grpc_cert", "", "the cert to use to connect")
go/vt/vttablet/grpctabletconn/conn.go:  key  = flag.String("tablet_grpc_key", "", "the key to use to connect")
go/vt/vttablet/grpctabletconn/conn.go:  ca   = flag.String("tablet_grpc_ca", "", "the server ca to use to validate servers when connecting")
go/vt/vttablet/grpctabletconn/conn.go:  crl  = flag.String("tablet_grpc_crl", "", "the server crl to use to validate server certificates when connecting")
go/vt/vttablet/grpctabletconn/conn.go:  name = flag.String("tablet_grpc_server_name", "", "the server name to use to validate server certificate")
```

### LINKED ISSUE #10975: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/grpctmclient`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/grpctmclient/*.go
go/vt/vttablet/grpctmclient/cached_client.go:   defaultPoolCapacity = flag.Int("tablet_manager_grpc_connpool_size", 100, "number of tablets to keep tmclient connections open to")
go/vt/vttablet/grpctmclient/client.go:  concurrency = flag.Int("tablet_manager_grpc_concurrency", 8, "concurrency to use to talk to a vttablet server for performance-sensitive RPCs (like ExecuteFetchAs{Dba,AllPrivs,App})")
go/vt/vttablet/grpctmclient/client.go:  cert        = flag.String("tablet_manager_grpc_cert", "", "the cert to use to connect")
go/vt/vttablet/grpctmclient/client.go:  key         = flag.String("tablet_manager_grpc_key", "", "the key to use to connect")
go/vt/vttablet/grpctmclient/client.go:  ca          = flag.String("tablet_manager_grpc_ca", "", "the server ca to use to validate servers when connecting")
go/vt/vttablet/grpctmclient/client.go:  crl         = flag.String("tablet_manager_grpc_crl", "", "the server crl to use to validate server certificates when connecting")
go/vt/vttablet/grpctmclient/client.go:  name        = flag.String("tablet_manager_grpc_server_name", "", "the server name to use to validate server certificate")
```

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This migrates the flags in `grpctabletconn` and `grpctmclient` to `pflag`, following the binary dependencies done in #10999 and #11066 respectively (with the exception of the addition of `vtbench`, which imports and uses `grpctabletconn` directly, rather than going indirectly through `tabletconn.Dialer()`.

## Related Issue(s)

- Closes #10974.
- Closes #10975.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/grpctabletconn/conn.go [hunk 1]
```diff
@@ -40,15 +41,29 @@ import (
...
+	name string
 )
 
+func registerFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&cert, "tablet_grpc_cert", cert, "the cert to use to connect")
+	fs.StringVar(&key, "tablet_grpc_key", key, "the key to use to connect")
+	fs.StringVar(&ca, "tablet_grpc_ca", ca, "the server ca to use to validate servers when connecting")
+	fs.StringVar(&crl, "tablet_grpc_crl", crl, "the server crl to use to validate server certificates when connecting")
+	fs.StringVar(&name, "tablet_grpc_server_name", name, "the server name to use to validate server certificate")
+}
```
Comment (GuptaManan100) [inline go/vt/vttablet/grpctabletconn/conn.go:57]: I am assuming these are the flags that are required on the binaries that have the grpc server side code

File: go/vt/vttablet/grpctmclient/client.go [hunk 2]
```diff
@@ -47,21 +46,46 @@ import (
...
 )
 
+func registerFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&concurrency, "tablet_manager_grpc_concurrency", concurrency, "concurrency to use to talk to a vttablet server for performance-sensitive RPCs (like ExecuteFetchAs{Dba,AllPrivs,App})")
+	fs.StringVar(&cert, "tablet_manager_grpc_cert", cert, "the cert to use to connect")
+	fs.StringVar(&key, "tablet_manager_grpc_key", key, "the key to use to connect")
+	fs.StringVar(&ca, "tablet_manager_grpc_ca", ca, "the server ca to use to validate servers when connecting")
+	fs.StringVar(&crl, "tablet_manager_grpc_crl", crl, "the server crl to use to validate server certificates when connecting")
+	fs.StringVar(&name, "tablet_manager_grpc_server_name", name, "the server name to use to validate server certificate")
+}
```
Comment (GuptaManan100) [inline go/vt/vttablet/grpctmclient/client.go:64]: And these are required by the binaries using client side code

**Review by GuptaManan100 — COMMENTED** [review]
I don't see vtcombo in the first set of binaries, but it is there in the second. Should'nt it be in both, since it uses both server side and client side grpc code

File: go/vt/vttablet/grpctabletconn/conn.go (hunk 1 above)
Comment (ajm188) [reply go/vt/vttablet/grpctabletconn/conn.go:57]: no, these are the flags that people that want to dial a `tabletconn` (client -> returns a `queryservice.QueryService`) need

File: go/vt/vttablet/grpctmclient/client.go (hunk 2 above)
Comment (ajm188) [reply go/vt/vttablet/grpctmclient/client.go:64]: and these are for people that want to make `tabletmanager` RPCs

**Comment by ajm188** [comment]
> I don't see vtcombo in the first set of binaries, but it is there in the second. 

`vtcombo` doesn't use the `queryservice` API, so it doesn't need `tabletconn` (or any implementations thereof, including `grpctabletconn`):

```
➜  vitess git:(andrew/pflag/grpctabletclients) ✗ go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... ./go/vt/servenv | awk -F"|" '$2 ~ " vitess.io/vitess/go/vt/vttablet/grpctabletconn " {print $1}' | grep vtcombo
```

**Review by GuptaManan100 — APPROVED** [review]
Thank-you for the clarification

**Pull request merged**